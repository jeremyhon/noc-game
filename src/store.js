import { createStore } from 'redux'
import rootReducer from './reducer'

const startStore = (initialState) => {
  return createStore(
    rootReducer,
    initialState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
}

const store = startStore()

export default store
