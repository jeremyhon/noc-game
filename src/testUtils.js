import { mount, shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';

// enzyme MUI Test Helpers
// - https://github.com/callemall/material-ui/issues/4664

const muiTheme = getMuiTheme();

/**
* MuiMountWithContext
*
* For `mount()` full DOM rendering in enzyme.
* Provides needed context for mui to be rendered properly
* during testing.
*
* @param {obj}    node - ReactElement with mui as root or child
* @return {obj}   ReactWrapper (http://airbnb.io/enzyme/docs/api/ReactWrapper/mount.html)
*/
export const muiMount = node => mount(node, {
  context: { muiTheme },
  childContextTypes: { muiTheme: PropTypes.object },
});

/**
* MuiShallowWithContext
*
* For `shallow()` shallow rendering in enzyme (component only as a unit).
* Provides needed context for mui to be rendered properly
* during testing.
*
* @param {obj}     node - ReactElement with mui
* @return {obj}    ShallowWrapper (http://airbnb.io/enzyme/docs/api/ShallowWrapper/shallow.html)
*/
export const muiShallow = node => shallow(node, {
  context: { muiTheme },
  childContextTypes: { muiTheme: PropTypes.object },
});

// touch tap event Helpers

export const simulateTouchTap = (element) => {
  const node  = ReactDOM.findDOMNode(element.node)
  TestUtils.Simulate.touchTap(node)
}
