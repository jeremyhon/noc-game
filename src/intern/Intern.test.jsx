import React from 'react'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'

import { muiMount, muiShallow, simulateTouchTap } from '../testUtils'
import { startStore } from '../store'
import ConnectedIntern, { Intern } from './Intern'

let component

describe('Unconnected Intern', () => {
  let props

  beforeEach(() => {
    props = {
      deselectIntern: jest.fn(),
      interns: [],
      populateInterns: jest.fn(),
    }
  })

  it("matches the snapshot", () => {
    component = shallow(<Intern {...props} />)
    expect(component).toMatchSnapshot()
  })

  it("generates new interns on startup", () => {
    component = muiMount(<Intern {...props} />)
    expect(props.populateInterns).toHaveBeenCalled()
  })

  it("closes the drawer", () => {
    component = muiMount(<Intern {...props} />)
    component.instance().changeDrawerState(false, 'swipe')
    expect(props.deselectIntern).toHaveBeenCalled()
  })
})

describe('ConnectedIntern', () => {
  let store;
  beforeEach(() => {
    store = startStore()
  })

  it("matches the snapshot", () => {
    component = muiShallow(<ConnectedIntern store={store}/>)
    expect(component).toMatchSnapshot()
  })

  describe("with provider", () => {
    beforeEach(() => {
      jest.spyOn(store, 'dispatch')
      component = muiMount(
        <Provider store={store}>
          <ConnectedIntern/>
        </Provider>
      )
    })

    it("generates new interns on startup", () => {
      expect(store.dispatch).toHaveBeenCalledWith({
        type: 'POPULATE_INTERNS',
        payload: expect.anything(),
      })
      expect(component.find('.intern-card')).toBePresent()
    })

    it("opens the drawer", () => {
      const button = component.find('.intern-card').first().find('.match-button')
      simulateTouchTap(button)
      expect(store.dispatch).toHaveBeenCalledWith({
        type: 'SELECT_INTERN',
        payload: expect.anything(),
      })
    })
  })
})
