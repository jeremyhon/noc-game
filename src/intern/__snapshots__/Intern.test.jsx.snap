// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedIntern matches the snapshot 1`] = `
<Intern
  deselectIntern={[Function]}
  interns={Array []}
  populateInterns={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;

exports[`Unconnected Intern matches the snapshot 1`] = `
<div
  className="intern"
>
  <Drawer
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={false}
    openSecondary={true}
    swipeAreaWidth={30}
    width={250}
    zDepth={2}
  />
</div>
`;
